rules/DeadCode/Rector/TryCatch/RemoveDeadCatchRector.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DeadCode\Rector\TryCatch;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Throw_;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Nop;
+use PhpParser\Node\Stmt\TryCatch;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DeadCode\Rector\TryCatch\RemoveDeadCatchRector\RemoveDeadCatchRectorTest
+ */
+final class RemoveDeadCatchRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove dead catches', [new CodeSample(
+            <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        try {
+            // some code
+        } catch (RuntimeException $exception) {
+            throw new InvalidArgumentException($exception->getMessage());
+        } catch (Throwable $throwable) {
+            throw $throwable;
+        }
+    }
+}
+CODE_SAMPLE
+            ,
+            <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        try {
+            // some code
+        } catch (RuntimeException $exception) {
+            throw new InvalidArgumentException($exception->getMessage());
+        }
+    }
+}
+CODE_SAMPLE
+        )]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [TryCatch::class];
+    }
+
+    /**
+     * @param TryCatch $node
+     * @return TryCatch|null
+     */
+    public function refactor(Node $node)
+    {
+        $catches = $node->catches;
```

Comment:
early check if catches only one:

```suggestion
        $catches = $node->catches;
        if (count($catches) === 1) {
            return null;
        }
```